extensions/wikia/ArticleComments/ArticleCommentsHooks.class.php

```diff
@@ -37,4 +37,19 @@ public static function onAfterPageHeaderButtons( \Title $title, array &$buttons
 	public static function onLoadExtensionSchemaUpdates( DatabaseUpdater $updater ) {
 		$updater->addExtensionTable( 'article_comments', MWInit::getExtensionsDirectory() . '/wikia/ArticleComments/sql/article_comments.sql' );
 	}
+
+	public static function onCommentUndelete( Title $title, $user, $reason ) {
+		$dbKey = $title->getDBkey();
+		if ( strpos($dbKey, '@comment' ) === false ) {
```

Comment:
let's check also namespace: `in_array( MWNamespace::getSubject( $title->getNamespace() ), $wgArticleCommentsNamespaces )` so as not to handle Wall & friends